### DISCUSSION TITLE
python socket.io "Received packet PONG data None Sending packet PING data None"

### DISCUSSION THREAD

**Discussion by DoctorMaduka** [miguelgrinberg/python-socketio#1034] [Q&A]
I am trying to scrape and receive live data from a Lithuanian website. Before now, I have been using picture processing libraries like `open-cv` and `Pillow` to take screenshots and transcribe the results. This is inefficient because my VM has to be connected to for these libraries to work. I have no experience with WebSockets and  would like to read the WebSocket messages.

The site uses socket.io and not much is available online to read on. Please help. 

Site: [url:demo.tv](https://demo.betgames.tv/)

code:
```
url ="wss://demo-nodejs.betgames.tv/socket.io/?EIO=3"

import asyncio
import socketio

loop = asyncio.get_event_loop()
sio = socketio.AsyncClient(logger=True, engineio_logger=True)
start_timer = None


@sio.event
async def connect():
    print('connected to server')

@sio.on('my message')
def on_message(data):
    print(data)

async def start_server():
    await sio.connect(url, transports=["websocket"])
    await sio.wait()


if __name__ == '__main__':
    loop.run_until_complete(start_server())```

![Screenshot 2022-09-15 171704.png](https://user-images.githubusercontent.com/113614090/190473717-2aa06fbe-56dd-4015-8ff2-8b528c2c41f0.png)

**Comment by DoctorMaduka** [comment]


**Reply by miguelgrinberg** [reply]
There are no indications of problems in this log. Your client application connects and then doesn't do anything more. That's what the logs reflect. The ping/pong messages are internal of the Socket.IO protocol, you can ignore them.

**Reply by miguelgrinberg** [reply]
The logs that you have will print all those messages, but you have to figure out how to make the server send them to you. The browser application knows how to do that. Your Python client apparently does not. This is not a problem with the Socket.IO protocol, it is that you need to know how to communicate with your server.

**Reply by DoctorMaduka** [reply]
Forgive me for I have a hard time understanding and Im pressed for time. Two more questions please, is it possible to handle eio message without using Auxiliary class, if not which Auxiliary class should I use?
```
import time
import socketio
import logging

url = "wss://demo-nodejs.betgames.tv/socket.io/?EIO=3"

sio = socketio.Client(logger=True, engineio_logger=True)


def main():
    logging.basicConfig(level=logging.DEBUG,
                        format='%(asctime)s %(levelname)-12s %(threadName)-10s %(funcName)15s() : %(message)s')
    info = {'stop': False}
    while True:
        try:
            logging.debug('login')
            time.sleep(0.75)
        except KeyboardInterrupt:
            info['stop'] = True
            break


@sio.on('my message')
def on_message(data):
    print('I received a message!')


@sio.event
def connect():
    print('connected to server')

    if sio.connected:
        main()


if __name__ == '__main__':
    sio.connect(url, transports=["websocket"])
    sio.wait()
```
![Screenshot 2022-09-16 213755](https://user-images.githubusercontent.com/113614090/190718639-dbb95cd2-b0c4-4c97-a3fe-36ddf839f4b0.png)

**Reply by miguelgrinberg** [reply]
I don't know what do you mean by auxiliary classes here. Also confused about you discussing EIO messages, which are a low-level aspect of the Socket.IO protocol and do not need to be handled by the application.

My impression is that all these questions come from the fact that you are not familiar with how Socket.IO works. There are many example servers and clients in this repository. I recommend that you start from one of those examples and once you have something working you start making changes to adapt it to your needs.

**Reply by miguelgrinberg** [reply]
This package has official documentation and a variety of examples, for those who are willing to learn Socket.IO from them. I hope you agree that it isn't fair of you to ask me to provide basic information when this information is available in docs and examples. Aside from that, if you are trying to integrate with a third-party Socket.IO web server, then I suggest you obtain specific information from the people who created the server.

**Reply by Notadumdum** [reply]
Hey brother @DoctorMaduka i think you need to use javascript to write the code... i have been having this problem for a long time now... and i have been hella confused cause the code i always tried worked and returned a session id but it never passed in my requests... yesterday i saw someone saying we have to use javascript for issues relating to browsers and all that.. i dont know if it is gonna work but i feel it is.. so i wanna learn a thing or two in javascript before i rewrite the code.. anyway please drop an email or something so we'd be able to communicate incase any of us could get it working first... Discord name is better still..mine is
![Screenshot_20220917-184258_Discord](https://user-images.githubusercontent.com/105162586/190869691-d4342cd2-197a-4755-af2d-e15728dcddae.jpg)

And just now i checked the internet again and saw this
![Screenshot_20220917-183807_Google](https://user-images.githubusercontent.com/105162586/190869718-5b659db5-4513-44e5-9fb4-bef39943a25b.jpg)
